spatial/r2/vector.go

```diff
@@ -12,34 +12,72 @@ type Vec struct {
 }
 
 // Add returns the vector sum of p and q.
+func Add(p, q Vec) Vec {
+	return Vec{
+		X: p.X + q.X,
+		Y: p.Y + q.Y,
+	}
+}
+
+// Sub returns the vector sum of p and -q.
+func Sub(p, q Vec) Vec {
+	return Vec{
+		X: p.X - q.X,
+		Y: p.Y - q.Y,
+	}
+}
+
+// Scale returns the vector p scaled by f.
+func Scale(f float64, p Vec) Vec {
+	return Vec{
+		X: f * p.X,
+		Y: f * p.Y,
+	}
+}
+
+// Dot returns the dot product p·q.
+func Dot(p, q Vec) float64 {
+	return p.X*q.X + p.Y*q.Y
+}
+
+// Cross returns the cross product p×q.
+func Cross(p, q Vec) float64 {
+	return p.X*q.Y - p.Y*q.X
+}
+
+// Add returns the vector sum of p and q.
+//
+// DEPRECATED: use r2.Add.
```

Comment:
Since we are using modules and we are in v0.\*, we don't need to deprecate, although we have done this once before. If we deprecate, can you add a todo to remove these at the v0.10.0 release. For the floats breaking change I moved the old API into a deprecated.go file with a TODO note.

Reply:
I'll move these into a `deprecate.go` file.

keeping these around for a release allows mechanization of the migration (via the `rf` scripts I put in the commit messages)